Typo fix, plus many changes and additions

DurvalMenezes opened the pull request:
See commit messages for individual changes.

DurvalMenezes committed:
Renamed '%' to 'mod', and 'pct' to '%'

DurvalMenezes committed:
Fixed typo: 'Subtracy' -> 'Subtract'

DurvalMenezes committed:
Added 'd' command: drop the top of the stack (x)

DurvalMenezes committed:
Added command "fac" to Calculate factorial of x.

DurvalMenezes committed:
- Renamed "c" command (to clear the stack) to "clr", to free it for the speed of light constant;
- Added math and physics constants

marcopaganini commented on main.go:
```diff
@@ -156,13 +156,32 @@ func main() {
+		"%": {"Calculate x% of y", 2, false, func(a []float64) (float64, error) { return a[0] * a[1] / 100, nil }},
+
+        "fac": {"Calculate factorial of x", 1, false, func(a []float64) (float64, error) {
+            var x = uint64(a[0])
```
You can do x := uint64(a[0])

DurvalMenezes replied:
sure, implemented.

marcopaganini commented on main.go:
```diff
@@ -156,13 +156,32 @@ func main() {
+            if (x <= 0) {
+                return 0, errors.New("fact requires a positive number")
+            }
+            var fact = uint64(1)
```
Same here: fact := 1

DurvalMenezes replied:
sure thing. done.

marcopaganini commented on main.go:
```diff
@@ -156,13 +156,32 @@ func main() {
+                return 0, errors.New("fact requires a positive number")
+            }
+            var fact = uint64(1)
+            for ix := uint64(1); ix <= x; ix++ {
```
No need for uint64(1) here... Just 1

DurvalMenezes replied:
Sure, resolved.

DurvalMenezes replied:
Oh, just noticed that then *none* of the variables involved in the calculation would be uint64. Won't Golang implicitly declare them as normal int (which has a much smaller range, and perhaps will have issues with large factorials)?

marcopaganini replied:
In this case, just do fact := uint64(x)

DurvalMenezes replied:
Did some testing and the minimum set of uint64() that works is both `fact := uint64(x)` as you suggested, plus `ix := uint64(1)` on the next line; please see my next PR.

marcopaganini commented on main.go:
```diff
@@ -156,13 +156,32 @@ func main() {
+            }
+            var fact = uint64(1)
+            for ix := uint64(1); ix <= x; ix++ {
+			    fact *= ix
```
Indentation issues? Please run gofmt on the entire source.

DurvalMenezes replied:
done.

marcopaganini commented on main.go:
```diff
@@ -156,13 +156,32 @@ func main() {
+		"d": {"Drop top of stack (x)", 1, true, func(_ []float64) (float64, error) { return 0, nil }},
+
+        // math & physical constants
+        "pi":  {"The famous transcedental number", 0, false, func(_ []float64) (float64, error)    { return 3.1415926536, nil }},
```
Constants should be prefixed by something or uppercase. Maybe const.pi? or PI?
Also, instead of using the number, use the constant from math. For example, Math.Pi, Math.E, etc (https://cs.opensource.google/go/go/+/refs/tags/go1.23.1:src/math/const.go;l=11)

DurvalMenezes replied:
Done, all are now uppercase; also all use golang libraries except for "MOL" and "C" because, AFAICS, no golang library defines them.

marcopaganini commented on main.go:
```diff
@@ -156,13 +156,32 @@ func main() {
+        "fac": {"Calculate factorial of x", 1, false, func(a []float64) (float64, error) {
+            var x = uint64(a[0])
+            if (x <= 0) {
+                return 0, errors.New("fact requires a positive number")
```
s/fact/factorial

DurvalMenezes replied:
Done, changed it in the message.

DurvalMenezes committed:
Per @marcopaganini's suggestions:
- used `gofmt` to properly format the code;
- changed all constants to uppercase;
- restored "c" command to clear the stack (now that the speed-of-light constant has been changed to uppercase "C");
- refactored "fac" code to use the minimum necessary uint64() conversions.

DurvalMenezes committed:
Fixed an implicit conversion which golangci-lint was complaning about
